letta/agent.py

```diff
@@ -1333,6 +1334,50 @@ def retry_message(self) -> List[Message]:
         assert all(isinstance(msg, Message) for msg in messages), "step() returned non-Message objects"
         return messages
 
+    def get_context_window(self) -> ContextWindowOverview:
+        """Get the context window of the agent"""
+
+        system_prompt = self.agent_state.system  # TODO is this the current system or the initial system?
+        num_tokens_system = count_tokens(system_prompt)
+        core_memory = self.memory.compile()
+        num_tokens_core_memory = count_tokens(core_memory)
+
+        # Check if there's a summary message in the message queue
+        if (
+            self._messages[1].role == MessageRole.user
+            and isinstance(self._messages[1].text, str)
+            # TODO remove hardcoding
```

Comment:
@sarahwooders this is pretty cringe but I'm not sure necessarily how else to do it because we don't tag summary messages in any special kind of way

Reply:
Can we make this a constant maybe? How is it formatted in the context window?